### PR TITLE
[Fix] Use torch.empty for output in attention+quant fusion

### PR DESCRIPTION
**Pull request by elvischenv**
## Purpose

#26680 replaced `torch.zeros` with `torch.empty` when initializing attn output.
This PR apply the change to attn+quant fusion.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by ProExpertProg — APPROVED** [review]

**Comment by ProExpertProg** [comment]
@elvischenv the failure seems related, could you take a look?

**Review by elvischenv — COMMENTED** [review]
> So how does the infused code work without initializing the output buffer? Or does it get initialized?

It seems that infused + BF16 uninitialized output buffer works, but fused + FP8 uninitialized output buffer does not work. And only failed with this `533-128-40-8` specific problem size, others can pass.

I tried changing the batch size from `533` to `512` and it passed. Would that probably be an issue of `TRITON_ATTN`?

**Comment by ProExpertProg** [comment]
@elvischenv could you push a fix where we initialize to zeros for triton attn and empty for otherwise, and create a gh issue? That way we can merge this and let the IBM team continue to investigate

**Review by elvischenv — COMMENTED** [review]
@ProExpertProg I cannot reproduce the failure from local. Let's see if it can pass on pipeline.

**Review by BoyuanFeng — APPROVED** [review]

**Comment by ProExpertProg** [comment]
### With fusion

```
vllm serve RedHatAI/Meta-Llama-3.1-8B-FP8 --attention-backend=TRITON_ATTN --kv-cache-dtype fp8 -cc.pass_config.fuse_attn_quant=true cudagraph_mode=FULL -cc.splitting_ops=[]

local-completions (pretrained=RedHatAI/Meta-Llama-3.1-8B-FP8,base_url=http://0.0.0.0:8669/v1/completions,num_concurrent=50,max_retries=3), gen_kwargs: (None), limit: None, num_fewshot: 5, batch_size: auto
```

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.4829|±  |0.0138|
|     |       |strict-match    |     5|exact_match|↑  |0.4814|±  |0.0138|

### Without fusion
```
vllm serve RedHatAI/Meta-Llama-3.1-8B-FP8 --attention-backend=TRITON_ATTN --kv-cache-dtype fp8 -cc.pass_config.fuse_attn_quant=false cudagraph_mode=FULL -cc.splitting_ops=[]

local-completions (pretrained=RedHatAI/Meta-Llama-3.1-8B-FP8,base_url=http://0.0.0.0:8669/v1/completions,num_concurrent=50,max_retries=3), gen_kwargs: (None), limit: None, num_fewshot: 5, batch_size: auto

```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.4784|±  |0.0138|
|     |       |strict-match    |     5|exact_match|↑  |0.4784|±  |0.0138|
```
